### PR TITLE
`pyreverse`: use colorblind friendly default colors

### LINKED ISSUE #8251: Use better default colors for `pyreverse`
### Current problem

The current default colors used by `pyreverse` when specifying `--colorized` are not visually pleasing and probably also not very colorblind friendly.

### Desired solution

Replace the current default colors with:

```python
[
    "dodgerblue",
    "orange",
    "teal",
    "sienna",
    "orchid",
    "lightsalmon",
    "plum",
    "gray",
    "yellow",
    "steelblue",
]
```

### Additional context

See discussion in https://github.com/PyCQA/pylint/pull/8223

### PR DESCRIPTION
**Pull request by DudeNr33**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |

## Description

As discussed in the issue, with 3.0 we can now change the default color palette.

Closes #8251

### CODE REVIEW

**Review by Pierre-Sassoulas — APPROVED** [review]
I did not review the graphical result... Code's LGTM :smile:

**Comment by DudeNr33** [comment]
For reference, this is how the new colors would look like:

image

**Comment by Pierre-Sassoulas** [comment]
It looks fine with protanopia (no red) but it's a little hard with other form of colorblindness. 

No blue:

tritanopia (no blue)

No coolor:
achromatopia (no color at all)

No green:
deuteranopia (no green)

orange 2 and goldenrod 9 are not great in particular. This is not something that can be improvised. What were the color before that (maybe it's already taken into account by plantuml) ? Otherwise some research are in order I think :D

**Comment by DudeNr33** [comment]
Oh that's cool, what tool did you use to make the different versions? Or did you just desaturate it yourself?

I tried to adapt the [colorblind palette from seaborn](https://seaborn.pydata.org/tutorial/color_palettes.html#qualitative-color-palettes) with the closest matching HTML colors. 
But I will try to find something better. 👍

**Comment by Pierre-Sassoulas** [comment]
It's firefox dev tools for accessibility (ctrl + shift + I), but there's probably the same thing in chrome. If it's already a colorblind palette then I don't think we're going to do much better than that, this is a hard problem to solve. They probably made the better result for red color blind because it's the most prevalent, it can't be perfect.

**Comment by DanielNoord** [comment]
I am actually colorblind and these colours are indeed sub-optimal. Funilly enough the "no green" output Pierre gives works really well for my type of colour blindness but I don't know if it would work for others.

**Comment by Pierre-Sassoulas** [comment]
I've found three possible color palettes:
IBM's : https://lospec.com/palette-list/ibm-color-blind-safe (It's IBM but there's only six of them)
Paul Tol's : https://personal.sron.nl/~pault/  (7, up to 10)
Ban wong: https://www.nature.com/articles/nmeth.1618 (8)

What's the best one in your opinion @DanielNoord ? Also do you know what type of color blindness you have ? It seems we should target red/green rather than blue/yellow, because red/green is a lot more common.

**Comment by DanielNoord** [comment]
I have red/green, the second link works quite well for me!

**Comment by DanielNoord** [comment]
> All figures in the second one, or a particular one ? :)

All are quite good, but figure 1 is definitely the best!

**Comment by Pierre-Sassoulas** [comment]
I think we should use figure 7 then, there's 9 different values and the contrast with the black text is still high enough (compared to say the dark blue in figure 4 where there is more values but I can't read the value in it personally).

fig7

Let me know if you want me to translate that in plantuml color @DudeNr33 :)

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
(Colorblindness to take into account)

**Comment by DudeNr33** [comment]
I used the hex codes directly and made the `PlantUMLPrinter` work with both hex codes and CSS names.

Here is an example:
image

Updated the functional test file to make sure that both CSS names and HEX values can be given through the command line.

File: pylint/pyreverse/main.py [hunk 1]
```diff
@@ -35,23 +35,15 @@
...
-    "mediumspringgreen",
+    "#77AADD",  # light blue
+    "#99DDFF",  # light cyan
+    "#44BB99",  # mint
+    "#BBCC33",  # pear
+    "#AAAA00",  # olive
+    "#EEDD88",  # light yellow
+    "#EE8866",  # orange
+    "#FFAABB",  # pink
+    "#DDDDDD",  # pale greay
```
Comment (Pierre-Sassoulas) [inline pylint/pyreverse/main.py:46]:
```suggestion
    "#DDDDDD",  # pale grey
```

File: pylint/pyreverse/main.py [hunk 2]
```diff
@@ -35,23 +35,15 @@
...
-    "cornflowerblue",
-    "cyan",
-    "darkgoldenrod",
-    "darkseagreen",
-    "dodgerblue",
-    "forestgreen",
-    "gold",
-    "hotpink",
-    "mediumspringgreen",
+    "#77AADD",  # light blue
```
Comment (Pierre-Sassoulas) [inline pylint/pyreverse/main.py:38]:
```suggestion
    # colorblind friendly palette taken from... (sorry I'm on mobile)
    "#77AADD",  # light blue
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Amazing we even got an enhancement for hexcode !

File: pylint/pyreverse/main.py (hunk 1 above)
Comment (DudeNr33) [reply pylint/pyreverse/main.py]: Oopsie

**Review by Pierre-Sassoulas — DISMISSED** [review]
👌

**Review by Pierre-Sassoulas — APPROVED** [review]

**Comment by nickdrozd** [comment]
I just saw the new colors -- they look great! :art:

**Pull request merged**